wperf/arg-parser-arg.cpp

```diff
@@ -0,0 +1,219 @@
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include "arg-parser-arg.h"
+#include <stdexcept>
+
+namespace ArgParserArg {
+    arg_parser_arg::arg_parser_arg(
+        const std::wstring name,
+        const std::vector<std::wstring> alias,
+        const std::wstring description,
+        const std::vector<std::wstring> default_values,
+        const int arg_count
+    ) : m_name(name), m_aliases(alias), m_description(description), m_arg_count(arg_count), m_values(default_values) {};
+
+    inline bool arg_parser_arg::operator==(const std::wstring& other_arg) const
+    {
+        return is_match(other_arg);
+    }
+
+    bool arg_parser_arg::is_match(const std::wstring& other_arg) const
+    {
+        return !other_arg.empty() && (other_arg == m_name || std::find(m_aliases.begin(), m_aliases.end(), other_arg) != m_aliases.end());
+    }
+
+    std::wstring arg_parser_arg::get_help() const
+    {
+
+        return L"\t" + get_all_flags_string() + L"\n" + arg_parser_add_wstring_behind_multiline_text(arg_parser_format_string_to_length(m_description), L"\t   ");
+    }
+
+    std::wstring arg_parser_arg::get_all_flags_string() const
+    {
+        if (get_alias_string().empty()) return m_name;
+        return m_name + L", " + get_alias_string();
+    }
+
+    std::wstring arg_parser_arg::get_usage_text() const
+    {
+        return m_description;
+    }
+
+    arg_parser_arg arg_parser_arg::add_alias(std::wstring new_alias)
+    {
+        m_aliases.push_back(new_alias);
+        return *this;
+    }
+
+    size_t arg_parser_arg::get_arg_count() const
+    {
+        return m_arg_count;
+    }
+
+    std::wstring arg_parser_arg::get_name() const
+    {
+        return m_name;
+    }
+
+    std::wstring arg_parser_arg::get_alias_string() const
```

Comment:
Maybe you can reuse:
```cpp
std::wstring WStringJoin(const std::vector<std::wstring>& input, std::wstring sep)
```

from `utils.cpp` ?

Reply:
Done